src/js/api/errors.ts

```diff
@@ -161,18 +165,23 @@ export const MSG_TECHNICAL_ERROR = 'technicalError';
  * @param {sourceError} [Error] - The lower level error, caught by the player, which resulted in this error.
  */
 export class PlayerError {
```

Comment:
can we use this class as the type when implementing or should we create a separate interface?

Reply:
I'm pretty sure we can use it as the type: [#TS Class - Advanced Techniques](https://www.typescriptlang.org/docs/handbook/classes.html#advanced-techniques)

> When you declare a class in TypeScript, you are actually creating multiple declarations at the same time. The first is the type of the instance of the class.

They also have an example on that page.